packages/sdk-cli/src/commands/mockHost.ts

```diff
@@ -26,8 +30,6 @@ export default function mockHost(cli: vorpal) {
     )
     .option('--maxAPIVersion <version>', 'Set the advertised max API version')
     .action(args => mockHostAction(hostType, args))
-    .cancel(() => {
-      if (socket) socket.close();
-    });
+    .cancel(() => handleCancel());
```

Comment:
can just be `.cancel(handleCancel);`

Reply:
TS warns it's not assigned yet, which at that point is true.

Reply:
Ahh fair.

Reply:
No, it can't. `.cancel(handleCancel)` will use the value of `handleCancel` at the time the command is registered, which is `undefined`. The anonymous function is needed so that _the value of `handleCancel` at the time of the event_ is called.